### PR TITLE
Add option to bypass OAuth for review apps

### PR DESCRIPTION
**Pull request by patch0**
## Status
- Part-implements RaspberryPiFoundation/profile#945 (for rails apps)

## Points for consideration:

- Security: Loglines are printed at the start of the app to show that the bypass is in use.  There is no further output.  I'd have liked to get a logline printed each time auth is attempted to show that the bypass is in use, but I wasn't able to cajole things into behaving.

## What's changed?

* A refinement as been added the OmniAuth::Configuration class to enable test mode, and set up dummy data.
* Dummy data is set by default but can be changed (see README.md)

## Steps to perform after deploying to production

The README.md describes the steps required to implement this in a rails app.  Broadly that goes something like the following added to your OmniAuth initialization.

```ruby
if ENV.has_key? 'BYPASS_AUTH' # This environment variable can be changed to whatever you like.
  using RpiAuthBypass
  OmniAuth.config.enable_rpi_auth_bypass
end
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,3 +42,42 @@ use OmniAuth::Builder do
     }
  )
 ```
+
+## Bypassing OmniAuth/OAuth
+
+It is also possible to bypass omniauth/ entirely, which can be useful in circumstances where hostnames are dynamic, e.g. in review deployments.  To do this add the following code to your OmniAuth initializer.
+
+```ruby
+if ENV.has_key? 'BYPASS_AUTH' # This environment variable can be changed to whatever you like.
```
Comment (glenpike) [inline README.md:51]: Just as a note - we've been using BYPASS_OAUTH in a couple of places - not sure if we need the env name to be more specific?

Comment (patch0) [reply README.md:51]: It can be whatever you like.  The variable isn't used in this gem at all.

Comment (patch0) [reply README.md:51]: Also it isn't specific to OAuth -- **all** auth processed by omniauth is bypassed.

**Comment by grega** [comment]
This is great - a really helpful addition! I guess the next step is to make sure we're using it... (Code Club, the voucher app and potentially the showcase submissions app being the initial use-cases for this...)

**Pull request merged**